### PR TITLE
[v1.4] Pull changes from Nukkit and support for 1.16.220

### PR DESCRIPTION
**Pull request by joserobjr**
This pulls recent commits from Cloudburst Nukkit and adds supports for 1.16.200.

This PR also contains code from #1040 , created by @Gabriel8579

### CODE REVIEW

**Review by joserobjr — COMMENTED** [review]
Self-reviewed, adjustments in next commit.

File: src/main/java/cn/nukkit/Player.java [hunk 1]
```diff
@@ -3212,33 +3210,18 @@ public void onCompletion(Server server) {
...
-                            if (itemDrop.getId() != Item.AIR) {
-                                vector3 = this.temporalVector.setComponents(itemFrame.x + 0.5, itemFrame.y, itemFrame.z + 0.5);
-                                this.level.dropItem(vector3, itemDrop);
-                                itemFrame.setItem(new ItemBlock(Block.get(BlockID.AIR)));
-                                itemFrame.setItemRotation(0);
-                                this.getLevel().addLevelEvent(this, LevelEventPacket.EVENT_SOUND_ITEM_FRAME_ITEM_REMOVED);
-                            }
-                        } else {
-                            itemFrame.spawnTo(this);
-                        }
```
Comment (Gabriel8579) [inline src/main/java/cn/nukkit/Player.java:3237]: I couldn't find the ItemFrameDropItemEvent() event call, is that correct?

**Review by Gabriel8579 — CHANGES_REQUESTED** [review]

File: src/main/java/cn/nukkit/blockentity/BlockEntityItemFrame.java [hunk 2]
```diff
@@ -120,10 +121,24 @@ public int getAnalogOutput() {
         return this.getItem() == null || this.getItem().getId() == 0 ? 0 : this.getItemRotation() % 8 + 1;
     }
 
+    @Since("1.4.0.0-PN")
+    public boolean dropItem(Player player) {
+        Item before = this.getItem();
+        if (before == null || before.isNull()) {
```
[the remark this answers is not in the document]
Comment (joserobjr) [reply src/main/java/cn/nukkit/blockentity/BlockEntityItemFrame.java:127]: Yes they added this method, it's no longer PowerNukkitOnly. Thanks for asking!

File: src/main/java/cn/nukkit/Player.java (hunk 1 above)
Comment (joserobjr) [reply src/main/java/cn/nukkit/Player.java:3237]: It was moved to `BlockEntityItemFrame.dropItemAndGetEntity(Player)`. Thank you for reviewing!

**Review by joserobjr — COMMENTED** [review]
Self-reviewed

**Pull request merged**